### DISCUSSION TITLE
Monitor `yarn.lock` to trigger reloads

### DISCUSSION THREAD

**Discussion by rauchg** [vercel/next.js#64629] [Ideas]
Not sure if we're ignoring it right now, but I've noticed that after running `yarn`, reloading is not triggered.

I think a smart shortcut to do that would be to detect `yarn.lock` having changed

**Comment by timneutkens** [comment]
@rauchg do you want to trigger a full process reload or a HMR?

**Comment by rauchg** [comment]
HMR. I suspect it should already trigger but maybe it doesn't ? Maybe cause it's non-js

**Comment by sedubois** [comment]
I think the same remark applies to `next.config.js`, changing it doesn't seem to trigger a reload.

**Comment by arunoda** [comment]
@sedubois I think we watch `next.config.js` and restart the webpack instance. We can't do HMR only for this because this might have core webpack config changes.

**Comment by sedubois** [comment]
@arunoda ok maybe, but as a user I don't see any message that changes in next.config.js are taken into account.

**Comment by timneutkens** [comment]
@arunoda @sedubois We do reload the server, also, it should show a message. https://github.com/zeit/next.js/blob/master/bin/next#L71-L81

Edit: just tested. It does show a message when reloading `next.config.js`

**Comment by sedubois** [comment]
@timneutkens I don't see that behaviour. When I change some text in `next.config.js` then hit save, Next doesn't give any message (such as `WAIT  Compiling...`) in console.

Don't know if it matters, but I run with `yarn dev` where `"dev": "nodemon -w server -w universal"`. Maybe the code you showed doesn't work with custom server.

**Comment by timneutkens** [comment]
It's using `next` indeed. For the custom server api you have to manage it yourself.

**Comment by sedubois** [comment]
OK, makes sense.

**Comment by rauchg** [comment]
New thought: yarn.lock deltas actually give us awesome information about _what has changed_, and we can invalidate the portions of the tree that are affected.

For example, if you have page `b.js` that depends on `react-portal`, and a dependency of `react-portal` changes, we would only schedule `b.js` for re-computation (specially in light of our upcoming re-factor to _only re-compute what's currently being rendered upon source change_

**Comment by timneutkens** [comment]
Looks like create-react-app does it this way: https://github.com/facebook/create-react-app/blob/next/packages/react-dev-utils/WatchMissingNodeModulesPlugin.js
